### PR TITLE
Bring over opal_progress_thread from master

### PR DESCRIPTION
**Pull request by jsquyres**
This is several commits from master: see the individual commit messages.

@rhc54 Note that I had to bring over open-mpi/ompi@836f49597d215fe1911bbb00e0ccad039611a61a, which removed the progress thread from ess_base_std_tool.c.  Please review the whole sequence, but with particular attention to this last commit.

Thanks.

### CODE REVIEW

**Comment by jsquyres** [comment]
When this PR is merged, it will close open-mpi/ompi#805.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/847/ for details.

**Comment by rhc54** [comment]
Will take a look this weekend

**Comment by hjelmn** [comment]
Once this is in I will bring over the openib btl cleanup.

**Comment by rhc54** [comment]
@jsquyres This just doesn't look right to me. The opal_progress_thread is correctly using the "sync" base, but the ORTE layer is also using "sync" instead of "async", and it looks like some unrelated stuff is included. Let me take a crack at it.

**Comment by rhc54** [comment]
@jsquyres Actually, looking at the commit messages, this is beginning to context-switch back - may actually be correct. Give me a little more time to grok it again.

**Comment by jsquyres** [comment]
Rebased to fix conflicts.

@rhc54 Got an ETA when you can review?  This PR is holding up the usnic update (#671).

File: orte/mca/ess/base/ess_base_std_tool.c [hunk 1]
```diff
@@ -243,7 +231,7 @@ int orte_ess_base_tool_finalize(void)
 
     /* release the event base */
     if (progress_thread_running) {
-        opal_stop_progress_thread("orte", true);
+        opal_progress_thread_finalize("orte");
```
Comment (rhc54) [inline orte/mca/ess/base/ess_base_std_tool.c]: I think this line should be removed - we removed the progress_thread create up above, so this "orte" thread will never be started in the tool.

Comment (jsquyres) [inline orte/mca/ess/base/ess_base_std_tool.c]: Fixed -- thanks!

**Comment by rhc54** [comment]
:+1:  - thanks!

**Comment by jsquyres** [comment]
@hppritcha Good to go

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/900/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/901/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ompi-release-v1.8-pr/903/ for details.

**Comment by lanl-ompi** [comment]
Build finished. No test results found.

**Comment by jsquyres** [comment]
Looks like LANL-discheck actually passed, but it failed to set the status back up here at Github.

**Comment by jsquyres** [comment]
@hppritcha ok'ed the merge.

**Pull request merged**